src/test/java/com/google/jenkins/plugins/computeengine/integration/ITUtil.java

```diff
@@ -0,0 +1,276 @@
+  private static final String ACCELERATOR_COUNT = "";
+  private static final String RUN_AS_USER = "jenkins";
+  static final String NULL_TEMPLATE = null;
+  private static final String NETWORK_NAME = format("projects/%s/global/networks/default");
+  private static final String SUBNETWORK_NAME = "default";
+  private static final boolean EXTERNAL_ADDR = true;
+  private static final String NETWORK_TAGS = "ssh";
+  private static final String SERVICE_ACCOUNT_EMAIL = "";
+  private static final String RETENTION_TIME_MINUTES_STR = "";
+  private static final String LAUNCH_TIMEOUT_SECONDS_STR = "";
+
+  static String format(String s) {
+    assertNotNull("GOOGLE_PROJECT_ID env var must be set", PROJECT_ID);
+    return String.format(s, PROJECT_ID);
+  }
+
+  static Credentials initCredentials(JenkinsRule r) throws Exception {
+    assertNotNull("GOOGLE_CREDENTIALS env var must be set", CREDENTIALS);
+    ServiceAccountConfig sac = new StringJsonServiceAccountConfig(CREDENTIALS);
+    Credentials credentials = new GoogleRobotPrivateKeyCredentials(PROJECT_ID, sac, null);
+
+    CredentialsStore store = new SystemCredentialsProvider.ProviderImpl().getStore(r.jenkins);
+    assertNotNull("Credentials store can not be null", store);
+    store.addCredentials(Domain.global(), credentials);
+    assertEquals(1, store.getCredentials(Domain.global()).size());
+    return credentials;
+  }
+
+  // Add Cloud plugin
+  static ComputeEngineCloud initCloud(JenkinsRule jenkinsRule) {
+    ComputeEngineCloud gcp =
+        new ComputeEngineCloud(null, CLOUD_NAME, PROJECT_ID, PROJECT_ID, "10", null);
+    assertEquals(0, jenkinsRule.jenkins.clouds.size());
+    jenkinsRule.jenkins.clouds.add(gcp);
+    assertEquals(1, jenkinsRule.jenkins.clouds.size());
+    return gcp;
+  }
+
+  static void handleClassLogs(StreamHandler streamHandler, String className) {
+    Logger logger = LogManager.getLogManager().getLogger(className);
+    assertNotNull(
+        String.format("Tried handling logs for %s which hasn't been instantiated yet.", className),
+        logger);
+    logger.addHandler(streamHandler);
+  }
+
+  // Get a compute client for out-of-band calls to GCE
+  static ComputeClient initClient(JenkinsRule jenkinsRule, Map<String, String> label, Logger log)
+      throws IOException {
+    ClientFactory clientFactory =
+        new ClientFactory(jenkinsRule.jenkins, new ArrayList<>(), PROJECT_ID);
+    ComputeClient client = clientFactory.compute();
+    assertNotNull("ComputeClient can not be null", client);
+    deleteIntegrationInstances(true, client, label, log);
+    return client;
+  }
+
+  static void teardown(
+      StreamHandler streamHandler,
+      ByteArrayOutputStream logOutput,
+      ComputeClient client,
+      Map<String, String> label,
+      Logger log)
+      throws IOException {
+    log.info("teardown");
+    deleteIntegrationInstances(false, client, label, log);
+    log.info(logs(streamHandler, logOutput));
```

Comment:
What is the motivation behind aggregating and nesting all the logs in this manner? I'd be concerned about this sort of nesting affecting downstream filtering capabilities.

Reply:
It turns out that this particular line was a remnant of my personal testing, when I was verifying the contents of the log output with method parallelization enabled, and it appears to just duplicate certain lines (those that are in the same thread as the base test).

The reason that logs were handled this way previously served two purposes:

1. Printing the contents of logOutput to the console in the event of an assertion failure.
2. Verifying the contents of logOutput in certain tests.

We could completely remove the log aggregation for these two reasons:

1. Could still be useful for us, although a more complete (involving methods from all threads and all classes/methods) version of the log output is created in the `target/failsafe-reports/` folder, but this may mean more effort to investigate assertion failures than having it printed to stderr.
2. I'm currently removing the tests which involve log contents and replacing them with stateful tests. So there's no need for this point anymore